spring-restdocs-asciidoctor/src/main/resources/extensions/srd-include-block-macro.rb

```diff
@@ -0,0 +1,103 @@
+require 'asciidoctor/extensions'
+require 'stringio'
+
+# Spring REST Docs block macro to import multiple snippet of an operation at once
+#
+# Usage
+#
+#   srd::operation-name[snippets='snippet-name1,snippet-name2', level=<indentation level>]
+#
+class RestDocsBlockMacro < Asciidoctor::Extensions::BlockMacroProcessor
+  use_dsl
+  named :std
+
+  def initialize name, config
+    super
+    # pre-defined section titles for commonly used snippets
+    @titles = {:'http-request' => 'HTTP request',
+               :'curl-request' => 'curl request',
+               :'httpie-request' => 'HTTPie request',
+               :'request-body' => 'Request body',
+               :'request-fields' => 'Request fields',
+               :'http-response' => 'HTTP response',
+               :'response-body' => 'Response body',
+               :'response-fields' => 'Response fields',
+               :'links' => 'Links'
+    }
+  end
+
+  def process(parent, reader, attrs)
+    doc = parent.document
+    snippet_dir = doc.attributes['snippets']
+    snippets = snippets_from_attributes(attrs)
+    section_level = section_level_from_attributes(attrs)
+
+    params = {:snippet_dir => snippet_dir,
+              :section_level => section_level,
+              :operation => reader}
+
+    content = StringIO.new
+    snippets.each do |snippet|
+      append_snippet_block(content, snippet, params)
+    end
+
+    add_snippets_block(content, doc, parent) unless content.length == 0
+    nil
+  end
+
+  def add_snippets_block(content, doc, parent)
+    fragment = Asciidoctor.load content,
+                                safe: doc.options[:safe],
+                                attributes: {'fragment' => '', 'projectdir' => doc.attr(:projectdir)}
+    fragment.blocks.each do |b|
+      b.parent = parent
+      parent << b
+    end
+  end
+
+  # if no section indentation level is provided, default to 4
+  def section_level_from_attributes(attrs)
+    (attrs['level'].to_i if attrs['level']) || 4
+  end
+
+  def snippets_from_attributes(attrs)
+    (attrs['snippets'].split(',') if attrs['snippets']) || []
```

Comment:
I think using [`fetch`](https://ruby-doc.org/core-1.9.3/Hash.html#method-i-fetch) would make this read a bit better:

```ruby
attrs.fetch('snippets', '').split(',')
```